libs/pilight/protocols/433.92/nexus.c

```diff
@@ -0,0 +1,257 @@
+/*
+	Copyright (C) 2014, 2018 CurlyMo & DonBernos & Michael Behrisch & Ligius
+
+	This Source Code Form is subject to the terms of the Mozilla Public
+	License, v. 2.0. If a copy of the MPL was not distributed with this
+	file, You can obtain one at http://mozilla.org/MPL/2.0/.
+	
+	Protocol info:
+	From https://forum.pilight.org/showthread.php?tid=3351&pid=22798#pid22798
+	ID ID ID CC         8-bit ID, the two least significant might encode the channel
+	BF CC               4 bits of flags:
+							B  =1 battery good
+							F  =1 forced transmission
+							CC =  channel, zero based
+	TT TT TT TT TT TT   12 bits signed integer, temp in Celsius/10
+	11 11               const / reserved
+	HH HH HH HH         8 bits, either 
+							- humidity (or 1111 xxxx if not available); or
+							- a CRC, e.g. Rubicson, algorithm in source code linked above
+
+	From: https://github.com/aquaticus/nexus433
+	Every bit starts with 500 µs high pulse. 
+	The length of the following low state decides if this is 1 (2000 µs) or 0 (1000 µs). 
+	There is a 4000 µs long low state period before every 36 bits.
+
+	See also https://github.com/merbanan/rtl_433/blob/master/src/devices/nexus.c
+*/
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "../../core/binary.h"
+#include "../../core/common.h"
+#include "../../core/dso.h"
+#include "../../core/gc.h"
+#include "../../core/log.h"
+#include "../../core/pilight.h"
+#include "../protocol.h"
+#include "nexus.h"
+
+#define PULSE_TOLERANCE 150
+#define MESSAGE_BITS 36
+#define RAW_LENGTH ((MESSAGE_BITS)*2) + 2
+
+enum PulseType {
+    START_P = 500,
+    ZERO_P = 1000,
+    ONE_P = 2000,
+    SYNC_P = 4000
+};
+
+typedef struct settings_t {
+    int id;
+    int channel;
+    int battery;
+    double temp;
+    double humi;
+    struct settings_t *next;
+} settings_t;
+
+static struct settings_t *settings = NULL;
+
+/**
+ * Validate whether a raw pulse length matches a known type
+ */
+static bool isValidPulse(uint16_t pulseLength, enum PulseType measureAgainst) {
+    return (pulseLength > (measureAgainst - PULSE_TOLERANCE)) && (pulseLength < (measureAgainst + PULSE_TOLERANCE));
+}
+
+static int validate(void) {
+    if (nexus->rawlen >= RAW_LENGTH && isValidPulse(nexus->raw[0], START_P) && isValidPulse(nexus->raw[RAW_LENGTH - 2], START_P)) {
```

Comment:
Small comment, but can you follow pilight style guide regarding the conditions.
`if(` instead of `if (`

Reply:
I just used CLang with the Google style, are there any other formatting errors? I will fix all of them

Reply:
Not errors, just preferences.

Reply:
Are there any other options to CLang? I have these: { BasedOnStyle: Google, IndentWidth: 4, ColumnLimit: 0}

Reply:
I don't now, i just use `notepad++` and manually type my code.

Reply:
I added SpaceBeforeParens: Never to my VSCode configuration, seems to the the job. Perhaps this should be documented into the "Contributing" section:
{ BasedOnStyle: Google, IndentWidth: 4, ColumnLimit: 0, SpaceBeforeParens: Never }

Reply:
I see a second PR comming :wink: